WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderTagStrip.java

```diff
@@ -0,0 +1,118 @@
+import org.wordpress.android.ui.reader.ReaderActivityLauncher;
+import org.wordpress.android.ui.reader.utils.ReaderUtils;
+import org.wordpress.android.util.AniUtils;
+import org.wordpress.android.util.DisplayUtils;
+
+/**
+ * used by the detail view to display the primary and secondary tags from a reader post
+ */
+public class ReaderTagStrip extends LinearLayout {
+
+    private LinearLayout mView;
+    private int mNumTags;
+    private String mAllTags;
+
+    public ReaderTagStrip(Context context) {
+        super(context);
+        initView(context);
+    }
+
+    public ReaderTagStrip(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        initView(context);
+    }
+
+    public ReaderTagStrip(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        initView(context);
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public ReaderTagStrip(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+        initView(context);
+    }
+
+    private void initView(Context context) {
+        mView = (LinearLayout) inflate(context, R.layout.reader_tag_strip, this);
+    }
+
+    public void setPost(@NonNull ReaderPost post) {
+        if (mView.getChildCount() > 0) {
+            mView.removeAllViews();
+        }
+
+        mNumTags = 0;
+
+        if (post.hasPrimaryTag()) {
+            addTag(post.getPrimaryTag());
+        }
+        if (post.hasSecondaryTag()) {
+            addTag(post.getSecondaryTag());
+        }
+
+        if (mNumTags == 0) {
+            mView.setVisibility(View.GONE);
+        } else if (mView.getVisibility() != View.VISIBLE) {
+            AniUtils.fadeIn(mView, AniUtils.Duration.SHORT);
+        }
+    }
+
+    private void addTag(final String tagName) {
+        String tagDisplayName = mNumTags > 0 ? ", " + ReaderUtils.makeHashTag(tagName) : ReaderUtils.makeHashTag(tagName);
+
+        // inflate a new textView to show this tag
+        LayoutInflater inflater = LayoutInflater.from(getContext());
+        TextView txtTag = (TextView) inflater.inflate(R.layout.reader_tag_strip_label, mView, false);
+        mView.addView(txtTag);
+
+        // skip showing this tag if it will make the view wider than the max
+        if (mNumTags > 0) {
+            // determine the width of the passed tag in the new textView
+            Rect rect = new Rect();
+            Paint paint = txtTag.getPaint();
+            paint.getTextBounds(tagDisplayName, 0, tagDisplayName.length(), rect);
+            int thisTagWidth = rect.width();
+
+            // determine the width of the tag already displayed
+            paint.getTextBounds(mAllTags, 0, mAllTags.length(), rect);
+            int currentTagWidth = rect.width();
+
+            int maxWidth = DisplayUtils.getDisplayPixelWidth(getContext());
+            if (currentTagWidth + thisTagWidth > maxWidth) {
```

Comment:
Maybe there is a better way to deal with long tags? By using display width we don't account for paddings/margins and date label length (it might be quite long). I made a couple of posts with secondary tags of various lengths - sometimes it's not visible at all, and sometimes it's ellipsized. We should probably show ellipsized second tag if there is at lest some room available for it.